leader-election-impl/src/main/java/com/palantir/paxos/PaxosQuorumChecker.java

```diff
@@ -116,23 +159,30 @@ private PaxosQuorumChecker() {
      * @param remotes a list of endpoints to make the remote call on
      * @param request the request to make on each of the remote endpoints
      * @param quorumSize number of acknowledge requests after termination
-     * @param executor runs the requests
+     * @param executors run the requests
      * @return a list of responses
      */
     private static <SERVICE, RESPONSE extends PaxosResponse> List<RESPONSE> collectResponses(
             ImmutableList<SERVICE> remotes,
             final Function<SERVICE, RESPONSE> request,
             int quorumSize,
-            Executor executor,
+            Map<SERVICE, ExecutorService> executors,
             long remoteRequestTimeoutInSec,
             boolean onlyLogOnQuorumFailure,
             boolean shortcircuitIfQuorumImpossible) {
-        CompletionService<RESPONSE> responseCompletionService = new ExecutorCompletionService<RESPONSE>(executor);
+        MultiplexingCompletionService<SERVICE, RESPONSE> responseCompletionService =
+                MultiplexingCompletionService.create(executors);
 
         // kick off all the requests
         List<Future<RESPONSE>> allFutures = Lists.newArrayList();
         for (final SERVICE remote : remotes) {
-            allFutures.add(responseCompletionService.submit(() -> request.apply(remote)));
+            try {
+                allFutures.add(responseCompletionService.submit(remote, () -> request.apply(remote)));
+            } catch (RejectedExecutionException e) {
+                if (shouldLogException()) {
+                    log.warn("Quorum checker executor rejected task", e);
```

Comment:
Non-blocking: could be nice to log some information about the executor as an unsafe arg for debugging purposes. Probably not this PR though as it's a bit large already